src/libxtp/jobcalculators/qmmm.cc

```diff
@@ -225,36 +239,55 @@ void QMMM::WriteJobFile(const Topology& top) {
 
   ofs << "<jobs>" << std::endl;
   Index jobid = 0;
-  std::string regionname = getFirstRegionName();
-  bool hasqm = hasQMRegion();
-  for (const Segment& seg : top.Segments()) {
-    for (const QMState& state : states_) {
-
-      std::string marker = std::to_string(seg.getId()) + ":" + state.ToString();
-      std::string tag = seg.getType() + "_" + marker;
-
-      tools::Property Input;
-      tools::Property& pInput = Input.add("input", "");
-      pInput.add("site_energies", marker);
-      tools::Property& regions = pInput.add("regions", "");
-      tools::Property& region = regions.add(regionname, "");
-      region.add("id", "0");
-      if (hasqm) {
-        region.add("state", state.ToString());
+  if (all_segments_) {
+    for (const Segment& seg : top.Segments()) {
+      for (const QMState& state : states_) {
+        Job job = createJob(seg, state, jobid);
+        job.ToStream(ofs);
+        jobid++;
+      }
+    }
+  } else {
+    for (Index segID : segments_) {
+      const Segment& seg = top.Segments()[segID];
+      for (const QMState& state : states_) {
+        Job job = createJob(seg, state, jobid);
+        job.ToStream(ofs);
+        jobid++;
```

Comment:
I would prefer we filled the segments_ vector completely for `all` at the beginning so we do not have to carry around the `all_segments_` bool and do not have to have an if here at all.